youpin.js

```diff
@@ -184,6 +184,48 @@ module.exports = (m, api, conversation, apiUserId) => {
         context.state    = STATE_WAIT_INTENT;
 
         greet( userid, context );
+				
+        // Move to the next state if the first message is a photo
+        if (attachments && !isSticker && (attachments[0].type == 'image' || attachments[0].type == 'video')) {
+          if (!context.isEnglish) {
+            m.sendText(userid, '(Y) แจ่มมากฮ่า');
+          } else {
+            m.sendText(userid, '(Y) Sweet!');
+          }
+          addPhotos(attachments, context);
+          setTimeout(() => {
+            context.lastSent = (new Date()).getTime();
+            context.state = STATE_WAIT_LOCATION;
+            if (!context.isEnglish) {
+              m.sendText(userid, `สวัสดีฮ่ะ คุณ ${profile.first_name} ช่วยพินสถานที่ที่พบปัญหา โดยการแชร์ ` +
+                'location จาก Messenger App บนมือถือของคุณด้วยฮ่า');
+              m.sendButton(userid, 'หรือมีอย่างอื่นให้ป้ายุพินช่วยจ๊ะ',
```

Comment:
You might need to wrap this with another `setTimeout((), 1000)` to ensure that the buttons are sent after the greeting message. 

BTW, I really like this solution 😄